### ISSUE TITLE
COLLECT on edge _to field doesn't group distinct values as expected.

### ISSUE THREAD

**Issue by aaronleesmith** [arangodb/arangodb#4827]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [ ] 2.8
- [ ] 3.0
- [ ] 3.1
- [ ] 3.2
- [x] 3.3
- [ ] self-compiled devel branch

Mode:
- [ ] Cluster
- [x] Single-Server

Storage-Engine:
- [x] mmfiles
- [ ] rocksdb

On this operating system:
- [ ] DCOS on
  - [ ] AWS
  - [ ] Azure
  - [ ] own infrastructure
- [ ] Linux 
  - [ ] Debian .deb
  - [ ] Ubuntu .deb
  - [ ] SUSE   .rpm
  - [ ] RedHat .rpm
  - [ ] Fedora .rpm
  - [ ] Gentoo
  - [x] docker - official docker library
  - [ ] other:
- [ ] Windows, version: 
- [ ] MacOS, version:

### this is an AQL-related issue:
[x] I'm using graph features

I'm issuing AQL via:
- [x] web interface with this browser:     running on this OS:
- [ ] arangosh
- [x] this Driver: arango java driver

I've run `db._explain("<my aql query>")` and it didn't shed more light on this.

The AQL query in question is:

```
FOR e IN Edge
    FILTER e._from == 'Vertex/source'
    COLLECT to=e._to INTO groups = {
      _key: e._key,
      _id: e._id
    }
    
    RETURN { to, groups }
```

The issue can be reproduced using this dataset:
https://www.dropbox.com/s/ds3enaxu3b2ptfh/dump.zip?dl=0

Please provide a way to create the dataset to run the above query on; either by a gist with an arangodump, or `db.collection.save({my: "values"}) statements. If it can be reproduced with one of the ArangoDB example datasets, it's a plus.

Unzip the dump, then

```
arangorestore --input-directory "dump"
```

These are the steps to reproduce:
1) open the browser on http://127.0.0.1:8529
2) log in as system
3) use database [ ] `_system` [x] other: test
4) run the sample query above on the "Queries" page

The following problem occurs: 

![screenshot 2018-03-12 12 45 51](https://user-images.githubusercontent.com/587826/37303330-c394f978-25f3-11e8-99e3-5e4d64cd0196.png)

Instead I would be expecting: 

There should be two rows of results, each showing all edges grouped by the distinct "e._to" ("dest, dest2").

If instead we group by e.type AND THEN e._to, the grouping works correctly.

**Comment by jsteemann** [comment]
Should be fixed by PR https://github.com/arangodb/arangodb/pull/4829

**Comment by aaronleesmith** [comment]
@jsteemann Thanks for addressing this so quickly.

**Comment by OmarAyo** [comment]
Hi @aaronleesmith  ,

Thanks for reporting this bug

We have released 3.3.5, and it includes a fix for it
I am closing this ticket now. Please feel free to reopen if the problem persists
